### PR TITLE
refactor: rename local API alias from a to api

### PR DESCRIPTION
**Pull request by lewis6991**
### Problem

Codebase inconsistently binds `vim.api` onto `a` or `api`.

### Solution
Use `api` everywhere. `a` as an identifier is too short to have at the module level.

### CODE REVIEW

**Review by clason — APPROVED** [review]

**Comment by clason** [comment]
Tests say you still missed some ;)

File: test/functional/terminal/scrollback_spec.lua [hunk 1]
```diff
@@ -607,12 +607,11 @@ describe("pending scrollback line handling", function()
   it("does not crash after nvim_buf_call #14891", function()
     skip(is_os('win'))
     exec_lua [[
-      local a = vim.api
-      local bufnr = a.nvim_create_buf(false, true)
-      a.nvim_buf_call(bufnr, function()
+      local bufnr = api.nvim_create_buf(false, true)
```
Comment (clason) [inline test/functional/terminal/scrollback_spec.lua:610]:
```suggestion
      local bufnr = vim.api.nvim_create_buf(false, true)
```

File: runtime/lua/vim/treesitter/query.lua [hunk 2]
```diff
@@ -74,7 +74,7 @@ end
 ---@return string[] query_files List of files to load for given query and language
 function M.get_files(lang, query_name, is_included)
   local query_path = string.format('queries/%s/%s.scm', lang, query_name)
-  local lang_files = dedupe_files(a.nvim_get_runtime_file(query_path, true))
+  local lang_files = dedupe_files(api.nvim_get_runtime_file(query_path, true))
```
Comment (clason) [inline runtime/lua/vim/treesitter/query.lua:77]:
```suggestion
  local lang_files = dedupe_files(vim.api.nvim_get_runtime_file(query_path, true))
```
Could inline this, as it's the only use in that file?

Comment (lewis6991) [reply runtime/lua/vim/treesitter/query.lua:77]: There were a couple `vim.api.*` that could use this.

Comment (clason) [reply runtime/lua/vim/treesitter/query.lua:77]: Maybe change (at least one of) those then to make it worth it? ;)

**Review by clason — APPROVED** [review]

File: runtime/lua/vim/treesitter/query.lua (hunk 2 above)
Comment (lewis6991) [reply runtime/lua/vim/treesitter/query.lua:77]: I did

Comment (clason) [reply runtime/lua/vim/treesitter/query.lua:77]: (sorry, only saw the reply on mail, not the actual changes)

**Pull request merged**